dimension_label.go

```diff
@@ -0,0 +1,405 @@
+//go:build experimental
+
+// This file declares Go bindings for experimental features in TileDB.
+// Experimental APIs to do not fall under the API compatibility guarantees and
+// might change between TileDB versions
+
+package tiledb
+
+/*
+#cgo LDFLAGS: -ltiledb
+#cgo linux LDFLAGS: -ldl
+#include <tiledb/tiledb_experimental.h>
+#include <tiledb/tiledb_serialization.h>
+#include <stdlib.h>
+*/
+import "C"
+import (
+	"fmt"
+	"reflect"
+	"runtime"
+	"unsafe"
+)
+
+type DimensionLabel struct {
+	tiledbDimensionLabel *C.tiledb_dimension_label_t
+	context              *Context
+}
+
+// Free releases the internal TileDB core data that was allocated on the C heap.
+// It is automatically called when this object is garbage collected, but can be
+// called earlier to manually release memory if needed. Free is idempotent and
+// can safely be called many times on the same object; if it has already
+// been freed, it will not be freed again.
+func (d *DimensionLabel) Free() {
+	if d.tiledbDimensionLabel != nil {
+		C.tiledb_dimension_label_free(&d.tiledbDimensionLabel)
+	}
+}
+
+// DimensionIndex returns the index of the dimension the dimension label provides labels for.
+func (d *DimensionLabel) DimensionIndex() (uint32, error) {
+	var dimensionIndex C.uint32_t
+	ret := C.tiledb_dimension_label_get_dimension_index(d.context.tiledbContext, d.tiledbDimensionLabel, &dimensionIndex)
+	if ret != C.TILEDB_OK {
+		return 0, fmt.Errorf("Error fetching dimension index for dimension label: %s", d.context.LastError())
+	}
+
+	return uint32(dimensionIndex), nil
+}
+
+// AttributeName returns the name of the attribute the label data is stored under.
+func (d *DimensionLabel) AttributeName() (string, error) {
+	var labelAttrName *C.char
+	ret := C.tiledb_dimension_label_get_label_attr_name(d.context.tiledbContext, d.tiledbDimensionLabel, &labelAttrName)
```

Comment:
Memory is owned by core but GoString at return makes a safe copy in the go space

Reply:
Can you add a comment saying this? Just so that future readers know why this is safe. (Similarly in other cases where we get a reference to C-owned memory that we copy out of.)